### PR TITLE
fix(connect): version check false positives

### PR DESCRIPTION
**Pull request by Lemonexe**
## Description

Attempt to fix and debug false positives in version check, caused by mismatch in responses from transport.

A new Sentry event should shed some more light as to why that happens (even though it's rare, it's high severity Connect bug).

## Notes for QA

N/A, there is no UI for this.

## Related Issue

Resolve https://github.com/trezor/trezor-suite-private/issues/185

🔍🖥️ **Suite native android test results:** [View in Currents](https://app.currents.dev/projects/iUe1Y4/runs?branch=fix/version-check-false-positives&lookback=7)

### CODE REVIEW

File: packages/connect/src/device/Device.ts [hunk 1]
```diff
@@ -930,6 +924,34 @@ export class Device extends TypedEmitter<DeviceEvents> {
...
+        if (!oldId || !!feat.bootloader_mode || !version) return;
+
+        const newVersion = getFirmwareOrBootloaderVersionArray(feat); // guaranteed to be FW mode here
+        // This should never happen, it's indicative of a transport-level bug, so log to Sentry via console.error
+        if (feat.device_id !== oldId) {
+            // transport descriptors are useful debug info, but no need to await, the side-effect to log to Sentry can run async
+            this.transport.enumerate().then(res => {
+                const descriptors = res.success ? res.payload : undefined;
+                const oldDevice = this.toMessageObject();
+                console.error('getFeatures response mismatched', oldDevice, feat, descriptors);
```
Comment (Lemonexe) [inline packages/connect/src/device/Device.ts:941]: In Suite, we can call imperatively `Sentry.captureException` and set `extraPayload` explciitely. In Connect we can only `console.error`, and I just found out that even with `console.error` this works to append payload to Sentry.

See [example thrown](https://satoshilabs.sentry.io/issues/7204404766/?project=5193825)  –  the [object Object] in the title is ugly, but we have the whole object in Additional data > Arguments 😮

**Review by mroz22 — APPROVED** [review]

File: packages/connect/src/device/Device.ts (hunk 1 above)
Comment (vojtatranta) [reply packages/connect/src/device/Device.ts:941]: the tests should have been udpated right? 

Since we limit emission here?

Comment (vojtatranta) [inline packages/connect/src/device/Device.ts:941]: NGL,  I needed to vibecode the proper response https://github.com/trezor/trezor-suite/pull/24566/changes

Comment (Lemonexe) [reply packages/connect/src/device/Device.ts:941]: nah not really, the event still emits in the case defined by the unit test. I added a new condition for the edge cases which are unit tested (maybe they should, I'll get on that), but the reason why the transport headers changed is unknown to me, but I'll find out.
